alternator/executor.cc

```diff
@@ -3680,6 +3680,23 @@ static size_t estimate_value_size(const rjson::value& value) {
 }
 
 class update_item_operation  : public rmw_operation {
+private:
+    struct update_context {
+        // The previous value of the item being updated, if read-before-write was performed.
+        const std::unique_ptr<rjson::value> previous_item;
+        // The desired timestamp of this operation.
+        const api::timestamp_type ts;
+
+        // The row being updated.
+        deletable_row& row;
+        // Attributes modified by this update.
+        attribute_collector modified_attrs;
+        // True if an attribute was updated.
+        bool any_updates;
```

Comment:
I wonder if instead of the apply_update() function setting any_updates=true, it wouldn't be clearer and cleaner for the caller to just set any_updates=true, and then this variable won't need to be in "state".
Same for any_delete.

(and then, when we don't have so much stuff in "state", I wonder if it needs to be a structure at all and can't just be passed as individual arguments)